theano/gof/tests/test_vm.py

```diff
@@ -438,3 +438,11 @@ def check_storage(storage_map):
         assert check_storage(storage_map)[0]
         assert len(set(id(v) for v in
                        itervalues(storage_map))) < len(storage_map)
+
+
+def test_no_recycling():
+    x = theano.tensor.vector()
+    mode = theano.Mode(optimizer='fast_compile')
```

Comment:
Why are we using `theano.Mode` inside `test_vm.py`?
Shouldn't we test variants of `VM_Linker`?